server/app/controllers/dev/seeding/DevDatabaseSeedController.java

```diff
@@ -93,11 +105,23 @@ public Result clear() {
     if (!isDevOrStaging) {
       return notFound();
     }
+    if (settingsManifest.getVersionCacheEnabled()) {
+      clearCache();
+    }
     resetTables();
     return redirect(routes.DevDatabaseSeedController.index().url())
         .flashing("success", "The database has been cleared");
   }
 
+  /** Remove all content from the cache. */
+  public void clearCache() {
+    if (!isDevOrStaging) {
+      return;
+    }
+    programsByVersionCache.removeAll();
+    questionsByVersionCache.removeAll();
```

Comment:
**suggestion (blocking):** call `join()` on these to ensure they complete before the request finishes, otherwise errors won't get surfaced to the caller

Reply:
Good call - updated!